src/ILLink.Tasks/LinkTask.cs

```diff
@@ -67,6 +67,24 @@ public class ILLink : ToolTask
 		/// </summary>
 		public string Warn { get; set; }
 
+		/// <summary>
+		/// Treat all warnings as errors.
+		/// Maps to '--warnaserror' if true, '--warnaserror-' if false.
+		/// </summary>
+		public bool TreatWarningsAsErrors { get; set; }
```

Comment:
Could you use a nullable bool like we do for some of the other boolean options? That way if the task is used without this setting, it won't be passed on the command-line - just to make it a little cleaner.